index.js

```diff
@@ -25,27 +25,53 @@ function compose (middleware) {
   }
 
   /**
+   * Make an iterator for middleware.
+   */
+
+  middleware[Symbol.iterator] = function () {
+    const self = this
+    const length = this.length
+    let i = 0
+    let context
+    let nextFunc
+
+    return {
+      next (c, n) {
+        if (!context) context = c
+        if (!nextFunc) nextFunc = n
+
+        let fn = self[i++] || nextFunc
+        let done = i > length
```

Comment:
Shouldn't this be `let done = i >= length + !!nextFunc`? It won't make a difference right now, but it will probably make a difference down the line.